include/deal.II/base/index_set.h

```diff
@@ -1683,6 +1681,17 @@ IndexSet::index_within_set (const size_type n) const
   p = Utilities::lower_bound(range_begin, range_end, r,
                              Range::end_compare);
 
+  // if we did not find, then
+  if (p==range_end)
```

Comment:
are you sure that all compilers will behave the same in that
```
if (p==range_end || p->end ==n || p->being >n)
```
they will never start dereference `range_end` here? I was not, that I split them into two `if`'s.